dataretrieval/wqp.py

```diff
@@ -91,7 +94,19 @@ def get_results(ssl_check=True, **kwargs):
     _warn_v3_profiles_outage()
 
     kwargs = _alter_kwargs(kwargs)
-    response = query(wqp_url('Result'), kwargs, delimiter=';', ssl_check=ssl_check)
+
+    if legacy is True:
+        warnings.warn('Legacy profiles return stale USGS data as of '
+                      'March 2024. Please set legacy=True to use the WQX3 '
+                      'data profiles and obtain the latest USGS data.')
+        url = wqp_url('Result')
+
+    else:
+        url = wqx3_url('Result')
+        if 'dataProfile' not in kwargs:
```

Comment:
Should check whether dataProfile is among a list of valid profiles and throw an error if not. Examples in `nwis`.